src/test/java/edu/ucsb/mapache/controllers/SearchHistoryControllerTests.java

```diff
@@ -0,0 +1,145 @@
+  @MockBean
+  GoogleSearchService googleSearchService;
+  @MockBean
+  AuthControllerAdvice authControllerAdvice;
+
+  @MockBean
+  private SearchRepository searchRepository;
+
+  @MockBean
+  private UserSearchRepository userSearchRepository;
+
+  @MockBean
+  private PropertiesService propertiesService;
+
+  private AppUser getMockAppUser() {
+    AppUser appUser = new AppUser();
+    appUser.setId(1);
+    appUser.setEmail("haixinlin123@umail.ucsb.edu");
+    appUser.setFirstName("Hunter");
+    appUser.setLastName("Lin");
+    appUser.setSearchRemain(100);
+    appUser.setTime(0);
+    appUser.setApiToken("fake-but-valid-api-token");
+    return appUser;
+  }
+
+  @Test
+  public void test_getJWT() {
+        DecodedJWT actual = searchhistorycontoller.getJWT(authorization);
+        // Writing the code twice is not the best way to do a test.
+        // In this case, I don't yet have a better alterantive to propose.
+        DecodedJWT expected = JWT.decode(authorization.substring(7));
+        // Note that the .equals for DecodedJWT is broken, so we need this instead:
+        assertEquals(expected.getToken(), actual.getToken());
+    }
+  
+  @Test
+  public void test_basicSearch_unauthorizedIfNotMember() throws Exception {
+    when(propertiesService.getNamespace()).thenReturn("https://proj-mapache-search.herokuapp.com");
+    mockMvc.perform(get("/api/members/searchhistory/allusersearches").contentType("application/json")
+        .header(HttpHeaders.AUTHORIZATION, exampleAuthToken)).andExpect(status().is(401));
+  }
+
+  @Test
+  public void test_quota_unauthorizedIfNotMember() throws Exception {
+
+    // TODO CHANGE TO APPROPRIATE URL FOR THIS CONTROLLER
+
+    // mockMvc
+    // .perform(
+    // get("/api/member/search/quota").contentType("application/json").header(HttpHeaders.AUTHORIZATION,
+    // exampleAuthToken))
+    // .andExpect(status().is(401));
+  }
+
+  @Test
+  public void test_getSearches_case1() throws Exception {
+
+    // write code to test the method
+    //
+    // @GetMapping("/allusersearches")
+    // public ResponseEntity<String> getSearches(@RequestHeader("Authorization")
+    // String authorization)
+
+    // I wrote 'case 1' because you'll probably need a few cases, e.g.
+    // for not logged in, logged in as admin, logged in as member, etc.
+    // depending on what the code is checking for
+  }
+
+  @Test
+  public void test_getSearches_case2() throws Exception {
+
+    // write code to test the method
+    //
+    // @GetMapping("/allusersearches")
+    // public ResponseEntity<String> getSearches(@RequestHeader("Authorization")
+    // String authorization)
+
+    // I wrote 'case 2' because you'll probably need a few cases, e.g.
+    // for not logged in, logged in as admin, logged in as member, etc.
+    // depending on what the code is checking for
+  }
+
+  // ETC... until you've tested all of the cases for all of the methods.
+  // Look in other controller tests for examples
```

Comment:
This code is NOT in final form to merge.   This is still essentially a draft PR.

Please let us know when it is ready to merge.  Code that's ready to merge doesn't have big blocks of commented out code in it.  This is code that I wrote to help your team... and you haven't done anything with it yet, at least not in this branch.